t/12-set_target_status.t

```diff
@@ -156,15 +156,15 @@ qq{
                 }
             })
             ngx.sleep(0.1) -- wait for initial timers to run once
-            local ok, err = checker:add_target("127.0.0.1", 2112, nil, true)
-            checker:set_target_status("127.0.0.1", 2112, false)
-            ngx.say(checker:get_target_status("127.0.0.1", 2112))  -- false
-            checker:report_http_status("127.0.0.1", 2112, 200)
-            checker:set_target_status("127.0.0.1", 2112, false)
-            checker:report_http_status("127.0.0.1", 2112, 200)
-            ngx.say(checker:get_target_status("127.0.0.1", 2112))  -- false
-            checker:report_http_status("127.0.0.1", 2112, 200)
-            ngx.say(checker:get_target_status("127.0.0.1", 2112))  -- true
+            local ok, err = checker:add_target("127.0.0.1", 2112, "127.0.0.1", true)
+            checker:set_target_status("127.0.0.1", 2112, "127.0.0.1", false)
+            ngx.say(checker:get_target_status("127.0.0.1", 2112, "127.0.0.1"))  -- false
+            checker:report_http_status("127.0.0.1", 2112, "127.0.0.1", 200)
+            checker:set_target_status("127.0.0.1", 2112, "127.0.0.1", false)
+            checker:report_http_status("127.0.0.1", 2112, "127.0.0.1", 200)
+            ngx.say(checker:get_target_status("127.0.0.1", 2112, "127.0.0.1"))  -- false
+            checker:report_http_status("127.0.0.1", 2112, "127.0.0.1", 200)
+            ngx.say(checker:get_target_status("127.0.0.1", 2112, "127.0.0.1"))  -- true
```

Comment:
It seems to me like all of those tests could have been updated with `nil` instead of a duplicate of the IP address. This way, we would also test for the backwards-compatibility of those functions (except for the new argument, of coursem, just the possibility of an argument being `nil` (e.g. `healthcheck_obj.hostname` in Kong or any application using this library).

Reply:
Fixed that, thanks!